bin/rank_roi

```diff
@@ -0,0 +1,273 @@
+
+
+class BrightestPixel(ROIFile):
+    """Pick region with brighest pixel
+    """
+    def compute_metric(self, propfile):
+        self.run_dmstat(propfile)
+        self.metric = float(self.dmstat.out_max)
+
+
+class FaintestPixel(ROIFile):
+    """Pick region with most faint max pixel
+    """
+    def compute_metric(self, propfile):
+        self.run_dmstat(propfile)
+        self.metric = -1.0*float(self.dmstat.out_max)
+
+
+class LargestArea(ROIFile):
+    """Pick region with most area
+    """
+    def compute_metric(self, propfile):
+        self.metric = self.roi.area()
+
+
+class SmallestArea(ROIFile):
+    """Pick region with least area
+    """
+    def compute_metric(self, propfile):
+        self.metric = -1.0*self.roi.area()
+
+
+def map_metric(func):
+    """Map string to object type"""
+    retval = {'min': LeastFlux, 'max': MostFlux,
+              'bright': BrightestPixel, 'faint': FaintestPixel,
+              'big': LargestArea, 'small': SmallestArea, }
+    return retval[func]
+
+
+def load_rois(pars):
+    'Load all roi files'
+    verb1("Loading ROI files")
+    import stk
+    rois = stk.build(pars["roifiles"])
+
+    metric = map_metric(pars["method"])
+    rois_reg = [metric(r, pars["outfile"]) for r in rois]
+    verb1("{} ROI files parsed".format(len(rois_reg)))
+    return rois_reg
+
+
+def clobber_outfiles(pars, rois_reg):
+    'Remove outfiles'
+    verb1("Checking clobber")
+    from ciao_contrib._tools.fileio import outfile_clobber_checks
+    pars["clobber"] = ("yes" == pars["clobber"])
+    for r in rois_reg:
+        outfile_clobber_checks(pars["clobber"], r.outfile)
+
+
+def compute_metric(pars, rois_reg):
+    'Compute roi metric'
+    verb1("Checking infile (image v. table)")
+    infile = pars["infile"]
+    oo = read_file(infile)
+    if isinstance(oo, IMAGECrate) is not True:
+        infile = infile+"[bin sky]"
+        verb1("Infile is a table, will bin into an image")
+    verb1("Computing metric for all ROIs")
+    for r in rois_reg:
+        r.compute_metric(infile)
+        verb2("region {}\tmetric {}".format(r.regionid, r.metric))
+
+
+def pick_winner(rois_reg):
+    'Determine which roi is best'
+    verb1("Determining which ROIs should get overlap area")
+    for r in rois_reg:
+        for k in rois_reg:
+            k.remove_from(r)
+
+
+def write_outputs(pars, rois_reg):
+    'Write output'
+    verb1("Writing out new roi files")
+    for r in rois_reg:
+        r.write(pars)
+
+
+@lw.handle_ciao_errors(toolname, __revision__)
+def main():
+    'main program'
+
+    from ciao_contrib.param_soaker import get_params
+    pars = get_params(toolname, "rw", sys.argv,
+                      verbose={"set": lw.set_verbosity, "cmd": verb1})
+
+    rois_reg = load_rois(pars)
+    clobber_outfiles(pars, rois_reg)
+    compute_metric(pars, rois_reg)
+    pick_winner(rois_reg)
+    write_outputs(pars, rois_reg)
+
+
+if __name__ == "__main__":
+    try:
+        main()
+    except Exception as E:
```

Comment:
This shouldn't be needed as it should be handled by the `handle_ciao_errors` decorator. Do you have an example where it is needed?

Reply:
I know I needed it at some point (it's my standard boiler-plate) ... I'm not known for adding a lot of extraneous error checking.  But what and under what circumstances I don't remember.

Reply:
You don't this logic in the other scrtipt, `random-region-tool-name-which-is-very-obvious-thank-you-jonathan`. I think.